src/passwordcheck.js

```diff
@@ -10,47 +10,71 @@ define(['jquery',
 function ($, bloomfilter) {
   'use strict';
 
-  var NO_OF_HASHING_FUNCTIONS = 8;
-  var bloomfilter = null;
-  $.ajax({
-    method: 'GET',
-    dataType: 'json',
-    url: '../bower_components/fxa-password-strength-checker/src/bloomdata_short_pwd.js',
-    error: function (jqxhr, status, err) { console.log(status + ' ERROR ' + err); }
-  }).done(function (res){
-    bloomfilter = new BloomFilter(res, NO_OF_HASHING_FUNCTIONS); //eslint-disable-line no-undef
-  });
-
-  function strengthCheck(password) {
-    // Check for passwords that at least contain a number and an alphabet,
-    // or if alphabets, then at least 12 characters long
-    var regex = new RegExp('((?=.*[a-zA-Z])(?=.*[0-9])[A-Za-z0-9!@#$%^&*()_+ ]{8,12})|([A-Za-z0-9!@#$%^&*()_+ ]{12,})');
-    return regex.test(password);
-  }
-
-  return function checkPasswordStrength(password, callback) {
-    if (password.length < 8) {
-      return;
-    }
-    var bool = false;
-    var strong = strengthCheck(password.toString());
-
-    if (strong) {
-      // Only if the password has a chance of being strong do we check with the bloom filter
-      // else, simply reject the password. This helps us to not store all-alpha or all-numeric passwords
-      // on the bloom filter, reducing space.
-      bool = bloomfilter.test(password);
-      callback('Password strength check with bloomfilter triggered');
-    } else {
-      bool = true;
-      callback('Password with all alphabets/numbers detected');
-    }
+  var URL = '../bower_components/fxa-password-strength-checker/src/bloomdata_short_pwd.js';
+
+  return function (options) {
+    // do something awesome with the options here, like set up the URL to fetch the data from, or
+    // set the minimum password length.
+    options = options || {};
+    var dataUrl = options.dataUrl || URL;
+    var minLength = options.minLength || 8;
+    var bloom = null;
+    var NO_OF_HASHING_FUNCTIONS = options.noOfHashFunctions || 8;
+    var errorCallBack = options.err || function (err) {
+        console.log(status + ' ERROR ' + err);
+      };
 
-    if (bool && strong) {
-      // password was found in the bloom filter, or was too weak.
-      callback('Password found in bloom filter');
-    } else if (!bool && strong) {
-      callback('Password was not found in bloom filter');
+    $.ajax({
```

Comment:
For a separate PR, since the content server is requiring the library asynchronously, making yet another XHR request to load the data is a bit overboard. I think a better approach would be to include grunt with this library and create a single bundle that can then be used in the content server and loaded asynchronously. Then we wouldn't have to worry about the bloom filter being available before using it, we could get rid of this XHR stuff, and passing in the alternative URL.